### PR TITLE
Pebble notice tempo ready

### PR DESCRIPTION
**Pull request by PietroPasotti**
This PR implements a pebble notice so that the charm can update its own status once the workload API becomes ready.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -190,6 +200,12 @@ def _requested_receivers(self) -> Tuple[ReceiverProtocol, ...]:
         requested_receivers = requested_protocols.intersection(set(self.tempo.supported_receivers))
         return tuple(requested_receivers)
 
+    def _on_tempo_pebble_custom_notice(self, event: PebbleNoticeEvent):
+        if event.notice.key == self.tempo.tempo_ready_notice_key:
+            logger.info("pebble api reports ready")
```
Comment (mmkay) [inline src/charm.py:205]: We could go down to debug with this one?

File: lib/charms/tempo_k8s/v2/tracing.py [hunk 2]
```diff
@@ -75,7 +75,6 @@ def __init__(self, *args):
     TYPE_CHECKING,
     Any,
     Dict,
-    Iterable,
     List,
```
Comment (mmkay) [inline lib/charms/tempo_k8s/v2/tracing.py:78]: Nit: should we bump lib versions for this or just assume it will get released with the next one?

**Review by mmkay — APPROVED** [review]
Couple of nitpicks, but overall looks good :)

File: lib/charms/tempo_k8s/v2/tracing.py (hunk 2 above)
Comment (PietroPasotti) [reply lib/charms/tempo_k8s/v2/tracing.py:78]: oh, I thought I'd rolled that back. Local tox -e lint complained about it.

**Pull request merged**